src/redux/initialState.ts

```diff
@@ -0,0 +1,110 @@
+import { Language } from '../component/LanguageSelector';
+import { DragPoint } from '../component/Node';
+import { Type } from '../config';
+import { AnyAction, FlowDefinition, Node, Position } from '../flowTypes';
+import { LocalizedObject } from '../services/Localization';
+
+export type Flows = Array<{ uuid: string; name: string }>;
+
+export interface ComponentDetails {
+    nodeUUID: string;
+    nodeIdx: number;
+    actionIdx?: number;
+    actionUUID?: string;
+    exitIdx?: number;
+    exitUUID?: string;
+    pointers?: string[];
+    type?: string;
+}
+
+export interface Components {
+    [uuid: string]: ComponentDetails;
+}
+
+export interface SearchResult {
+    name: string;
+    id: string;
+    type?: string;
+    prefix?: string;
+    extraResult?: boolean;
+}
+
+export interface ContactFieldResult extends SearchResult {
+    key?: string;
+}
+
+export interface CompletionOption {
+    name: string;
+    description: string;
+}
+
+export interface PendingConnections {
+    [uuid: string]: DragPoint;
+}
+
+export interface ReduxState {
```

Comment:
the component's initial state; includes some properties on the `ComponentMap` and `FlowMutator` singletons